vars/cloudFoundryDeploy.groovy

```diff
@@ -145,7 +152,14 @@ def deployCfNative (config) {
             export HOME=${config.dockerWorkspace}
             cf login -u \"${username}\" -p '${password}' -a ${config.cloudFoundry.apiEndpoint} -o \"${config.cloudFoundry.org}\" -s \"${config.cloudFoundry.space}\"
             cf plugins
-            cf ${deployCommand} ${config.cloudFoundry.appName?:''} ${config.deployType == 'blue-green'?'--delete-old-apps':''} -f '${config.cloudFoundry.manifest}' ${config.smokeTest}"""
+            cf ${deployCommand} ${config.cloudFoundry.appName?:''} ${blueGreenDeployOptions} -f '${config.cloudFoundry.manifest}' ${config.smokeTest}"""
+        if (config.keepOldInstance) {
+            sh """#!/bin/bash
+            set +x  
+            export HOME=${config.dockerWorkspace}
+            cf stop ${config.cloudFoundry.appName?:''}
```

Comment:
Elvis operator makes no sense here since the app name is mandatory

```
$ cf stop
Incorrect Usage: the required argument `APP_NAME` was not provided
```

So the app name is required for stop to be called and has to be moved into the if's expression

Reply:
But if app name is not there, we will fail the step before this.. I'd get rid of the elvis here, just makes the code more complex for no effect